### PR TITLE
chore: remove type comments from ddtrace/_trace/

### PR DESCRIPTION
**Pull request by brettlangdon**
Convert type hint comments to annotations in `ddtrace/_trace/`

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Comment by brettlangdon** [comment]
There are some meaty changes here, I can break up if that is helpful.

**Review by emmettbutler — APPROVED** [review]

**Comment by brettlangdon** [comment]
hmm.... pre_check is passing, but typing is failing on actual tests 🤔

**Review by emmettbutler — COMMENTED** [review]
I don't think the `if TYPE_CHECKING` approach works here, see the CI errors

File: ddtrace/debugging/_signal/tracing.py [hunk 1]
```diff
@@ -33,7 +33,7 @@
 class DynamicSpan(Signal):
     """Dynamically created span"""
 
-    _span_cm = attr.ib(type=t.Optional[t.ContextManager[Span]], init=False)
+    _span_cm = attr.ib(type=t.Optional[Span], init=False)
```
Comment (P403n1x87) [inline ddtrace/debugging/_signal/tracing.py:36]: Is this causing typing issues?

Comment (brettlangdon) [reply ddtrace/debugging/_signal/tracing.py:36]: yeah 🙄

**Review by P403n1x87 — APPROVED** [review]

File: ddtrace/_trace/span.py [hunk 2]
```diff
@@ -267,21 +259,18 @@ def finished(self, value):
             self.duration_ns = None
 
     @property
-    def duration(self):
-        # type: () -> Optional[float]
+    def duration(self) -> Optional[float]:
```
Comment (erikayasuda) [inline ddtrace/_trace/span.py:262]: Explicit type hinting added here

**Pull request merged**